flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NettyShuffleEnvironment.java

```diff
@@ -303,7 +309,18 @@ private void registerInputMetrics(MetricGroup inputGroup, MetricGroup buffersGro
 			InputGateMetrics.registerQueueLengthMetrics(inputGroup, inputGates);
 		}
 		buffersGroup.gauge(METRIC_INPUT_QUEUE_LENGTH, new InputBuffersGauge(inputGates));
-		buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new InputBufferPoolUsageGauge(inputGates));
+
+		if (config.isCreditBased()) {
```

Comment:
Note: in another PR #8485 the metrics operations are refactored in a metric factory. If that PR merged firstly then we might rebase to adjust this part.

Reply:
ok